src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.dao.impl.MovieDaoImpl;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
+    private final SessionFactory sessionFactory;
+    private final MovieDao movieDao;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+        this.movieDao = new MovieDaoImpl(sessionFactory);
+    }
+
     @Override
     public Movie add(Movie movie) {
-        return null;
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        return movieDao.get(id).orElseThrow(() ->
+                new RuntimeException("Movie not found"));
```

Comment:
Consider providing more detailed information in the exception message when a movie is not found, such as including the movie's ID. This can help in debugging and understanding which specific movie retrieval failed.